src/foraging_gui/TransferToNWB.py

```diff
@@ -317,91 +338,88 @@ class obj:
+                        delay_max=float(obj.TP_DelayMax[i]),
+                        delay_duration=obj.B_DelayHistory[i],
+                        ITI_beta=float(obj.TP_ITIBeta[i]),
+                        ITI_min=float(obj.TP_ITIMin[i]),
+                        ITI_max=float(obj.TP_ITIMax[i]),
+                        ITI_duration=obj.B_ITIHistory[i],
+                        response_duration=float(obj.TP_ResponseTime[i]),
+                        reward_consumption_duration=float(obj.TP_RewardConsumeTime[i]),
+                        auto_waterL=obj.B_AutoWaterTrial[0][i] if type(obj.B_AutoWaterTrial[0]) is list else obj.B_AutoWaterTrial[i],   # Back-compatible with old autowater format
+                        auto_waterR=obj.B_AutoWaterTrial[1][i] if type(obj.B_AutoWaterTrial[0]) is list else obj.B_AutoWaterTrial[i],
+                        # optogenetics
+                        laser_on_trial=obj.B_LaserOnTrial[i],
+                        laser_wavelength=LaserWavelengthC,
+                        laser_location=LaserLocationC,
+                        laser_1_power=Laser1Power,
+                        laser_2_power=Laser2Power,
+                        laser_on_probability=LaserOnProbablityC,
+                        laser_duration=LaserDurationC,
+                        laser_condition=LaserConditionC,
+                        laser_condition_probability=LaserConditionProC,
+                        laser_start=LaserStartC,
+                        laser_start_offset=LaserStartOffsetC,
+                        laser_end=LaserEndC,
+                        laser_end_offset=LaserEndOffsetC,
+                        laser_protocol=LaserProtocolC,
+                        laser_frequency=LaserFrequencyC,
+                        laser_rampingdown=LaserRampingDownC,
+                        laser_pulse_duration=LaserPulseDurC,
+                        
+                        session_wide_control=_get_field(obj, 'TP_SessionWideControl', index=i, default='None'),
+                        fraction_of_session=float(_get_field(obj, 'TP_FractionOfSession', index=i, default=np.nan)),
+                        session_start_with=_get_field(obj, 'TP_SessionStartWith', index=i, default='None'),
+                        session_alternation=_get_field(obj, 'TP_SessionAlternating', index=i, default='None'),
+                        # add all auto training parameters (eventually should be in session.json)
+                        auto_train_engaged=_get_field(obj, 'TP_auto_train_engaged', index=i,default='None'),
+                        auto_train_curriculum_name=_get_field(obj, 'TP_auto_train_curriculum_name', index=i, default='None'),
+                        auto_train_curriculum_version=_get_field(obj, 'TP_auto_train_curriculum_version', index=i, default='None'),
+                        auto_train_curriculum_schema_version=_get_field(obj, 'TP_auto_train_curriculum_schema_version', index=i, default='None'),
```

Comment:
@XX-Yin  Well, you changed the default from 'none' to 'None', which unfortunately broke the AutoTrain pipeline...

Next time, please be careful not to change the result of a code when you refactor it.

Reply:
To make sure the whole system is up and running before the next morning, I just spent another hour fixing everything, which is really frustrating...

I should have reviewed the code more carefully. At the same time, we should be 200% careful when our code has a huge impact.